raiden/transfer/architecture.py

```diff
@@ -29,7 +29,54 @@
 # outputs are separated under different class hierarquies (StateChange and Event).
 
 
-class State:
+class FlatInheritance(type):
+    """ Nested inheritance is forbidden for architecture classes.
+
+    The design choices adopted by this architecture rely heavily on dynamic
+    dispatching based on type checking, e.g.::
+
+        if isintance(obj, StopStateChange):
+            handle_stop(state, obj)
+        elif isintance(obj, StopWithCareStateChange):
+            handle_stop_with_care(state, obj)
+
+    For the above snippet to work, it's important for StopWithCareStateChange
+    to not be a subclass of StopStateChange, otherwise the order of the
+    statements used for the dynamic dispatch matter, leading to subtle bugs
+    like the one above.
+    """
+
+    def __new__(mcs, name, bases, namespace, **kwds):  # pylint: disable=unused-argument
```

Comment:
For consistency can you rename `**kwds` to `**kwargs`?

Also I have seen the `namespace` argument being called `attrs`. Can we also call it like that as I think it makes a bit more sense. It's the output of the `__prepare__` function right?

Note: on namespace naming not really strong opinion -- so if you prefer it like this I am fine.